yt_dlp/extractor/nhk.py

```diff
@@ -45,24 +47,32 @@ def _get_api_info(self, refresh=True):
         self.cache.store('nhk', 'api_info', api_info)
         return api_info
 
-    def _extract_formats_and_subtitles(self, vod_id):
+    def _extract_stream_info(self, vod_id):
         for refresh in (False, True):
             api_info = self._get_api_info(refresh)
             if not api_info:
                 continue
 
             api_url = api_info.pop('url')
-            stream_url = traverse_obj(
-                self._download_json(
-                    api_url, vod_id, 'Downloading stream url info', fatal=False, query={
-                        **api_info,
-                        'type': 'json',
-                        'optional_id': vod_id,
-                        'active_flg': 1,
-                    }),
-                ('meta', 0, 'movie_url', ('mb_auto', 'auto_sp', 'auto_pc'), {url_or_none}), get_all=False)
+            meta = traverse_obj(self._download_json(api_url, vod_id, 'Downloading stream url info',
+                                fatal=False, query={
+                                    **api_info,
+                                    'type': 'json',
+                                    'optional_id': vod_id,
+                                    'active_flg': 1, }), ('meta', 0))
+
+            stream_url = traverse_obj(meta, ('movie_url', ('mb_auto', 'auto_sp', 'auto_pc'),
+                                      {url_or_none}), get_all=False)
+
+            info = traverse_obj(meta, {
+                'duration': 'duration',
+                'timestamp': ('publication_date', {unified_timestamp}),
+                'release_timestamp': ('insert_date', {unified_timestamp}),
+                'modified_timestamp': ('update_date', {unified_timestamp}),
+            })
+
             if stream_url:
-                return self._extract_m3u8_formats_and_subtitles(stream_url, vod_id)
+                return self._extract_m3u8_formats_and_subtitles(stream_url, vod_id), info
```

Comment:
IMO let's transform all the data to an info_dict structure here in this function to make things more readable when we call it

```py
            if stream_url:
                formats, subtitles = self._extract_m3u8_formats_and_subtitles(stream_url, vod_id)
                return {
                    **traverse_obj(meta, {
                        'duration': ('duration', {int_or_none}),
                        'timestamp': ('publication_date', {unified_timestamp}),
                        'release_timestamp': ('insert_date', {unified_timestamp}),
                        'modified_timestamp': ('update_date', {unified_timestamp}),
                    }),
                    'formats': formats,
                    'subtitles': subtitles,
                }
```